\section{Heuristics approachesfor pairwise sequence alignment}
These methods are very popular because they are much faster than dynamic
programming and at the same time they return excellent alignment of biologica
sequences.

\subsection{General strategy}

The general strategy of these methods is based on two main steps:
\begin{enumerate}
  \item Produce a lookup table where the words of length $k$ (k-mers) present
in sequence A are indexed, including all the position where they occur in
sequence A.
  \item Read sequence B, word by word. For every word check in the lookup talbe
if and where the word occur in sequence A. Then try to extend the alignment.
\end{enumerate}

Other strategies are: FASTA, BLAST and BLAT.

\paragraph*{Conversion of DNA kmerso into numbers}

With a given DNA you can translate ACGT with a binary number. In this way you
can convert a DNA sequence into a number. This can be done also with ammino
acid.
But how do you fill up the metrics? This is why there are different metrics.

\subsection{FASTA}

Imagine a sequence placed at the top and a sequence placed on the left side,
both starting at the top left corner. The sequence at the top is indexed,
memoriing all the occurrences and position ofeach kmer. Then the sequence at
the left is read, one kmer at a time. All the "growing" diagonals are memorized
to produce the result shown in panel A.

For example, if you have kmer of length one ($ktup = 1$), you will find better
performance in confront of Smith \& Waterman. This because you have to do less
operation with FASTA.

You use Smith \& Waterman only if you go for a full match, because analize a
full alignment it's expensive but Smith \& Waterman it's more accurate.

\begin{figure}[H]
  \centering
  \includegraphics[scale=0.5]{FASTA}
  \caption{FASTA}
\end{figure}

Question: If I have a very long sequence and several short DNA sequence, is it
better to index the long or the short sequence? Reply: it's better to index
the long database, because you can check all the differents sequence in the
database where are all the data.

\subsection{BLAST}
